### PR TITLE
Search and filtering UI

### PR DESCRIPTION
**Pull request by mythmon**
- [x] Search, filter by enabled, and filter by action
- [x] Fetch action list from server instead of hard coding it
- [x] Link recipe query to the URL
- [x] Tests

### CODE REVIEW

**Comment by mythmon** [comment]
This is ready for review, but it depends on #497 and #512 to merge.

**Comment by mythmon** [comment]
With all the blockers merged, I've rebased this and it is ready for review.

File: content/components/recipes/details/RecipeQueryEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+  return (
+    <Form className={className} layout="inline">
+      <FormGroup controlId="text">
+        <ControlLabel>Search</ControlLabel>
+        <InputGroup>
+          <InputGroup.Addon>
+            <Icon icon="search" />
+          </InputGroup.Addon>
+          <FormControl
```
Comment (rehandalal) [inline content/components/recipes/details/RecipeQueryEditor.tsx:122]: I don't think a `FormControl` is supposed to be inside of an `InputGroup` and its causing a bit of a layout issue but I don't think that should block merging

**Review by rehandalal — APPROVED** [review]
bors r+

**Comment by rehandalal** [comment]
bors d=@mythmon

File: content/components/recipes/details/RecipeQueryEditor.tsx (hunk 1 above)
Comment (mythmon) [reply content/components/recipes/details/RecipeQueryEditor.tsx:122]: What is the layout issue you are seeing? It looks fine on my end.

image

**Pull request merged**